From 2cfec46a8f94c3576a228f1a603767824b2eda8c Mon Sep 17 00:00:00 2001
From: Richard Zhu <hongxing.zhu@nxp.com>
Date: Thu, 15 Jun 2017 11:41:34 +0800
Subject: [PATCH 1903/5242] MLK-15080 PCI: imx: pcie ep can't be probed
 properly

commit  b8c533ce5cda049debf82e9b191901b19a5cbfdc from
https://source.codeaurora.org/external/imx/linux-imx.git

iMX7D Sabre SD board implement the GPIO expander
connected to a peripheral bus.
Probe deferral would be triggered when try to request
the expanded GPIO at the first time.
pcie ep can't be probed properly at the second probe,
because of the duplicated registration of the sysfs.

Change the registeration point of the sysfs to fix
this issue.

Signed-off-by: Richard Zhu <hongxing.zhu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/pci/controller/dwc/pci-imx6.c |   12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/drivers/pci/controller/dwc/pci-imx6.c b/drivers/pci/controller/dwc/pci-imx6.c
index dd8f869..7fab522 100644
--- a/drivers/pci/controller/dwc/pci-imx6.c
+++ b/drivers/pci/controller/dwc/pci-imx6.c
@@ -1226,13 +1226,6 @@ static int __init imx6_pcie_probe(struct platform_device *pdev)
 	imx6_pcie->variant =
 		(enum imx6_pcie_variants)of_device_get_match_data(dev);
 
-	if (IS_ENABLED(CONFIG_EP_MODE_IN_EP_RC_SYS)) {
-		/* add attributes for device */
-		ret = sysfs_create_group(&pdev->dev.kobj, &imx_pcie_attrgroup);
-		if (ret)
-			return -EINVAL;
-	}
-
 	np = of_find_compatible_node(NULL, NULL, "fsl,imx-pcie-phy");
 	if (np != NULL) {
 		imx6_pcie->phy_base = of_iomap(np, 0);
@@ -1426,6 +1419,11 @@ static int __init imx6_pcie_probe(struct platform_device *pdev)
 		LIST_HEAD(res);
 		struct resource_entry *win, *tmp;
 
+		/* add attributes for device */
+		ret = sysfs_create_group(&pdev->dev.kobj, &imx_pcie_attrgroup);
+		if (ret)
+			return -EINVAL;
+
 		ret = of_pci_get_host_bridge_resources(np, 0, 0xff, &res,
 						       &pp->io_base);
 		if (ret)
-- 
1.7.9.5

